roles/pulp_devel/tasks/install_podman.yml

```diff
@@ -45,20 +45,37 @@
   ignore_errors: '{{ ansible_facts.distribution_version == "8.1" }}'
 
 - name: Adding podman registries (Debian-specific)
-  command: echo -e "[registries.search]\nregistries = ['docker.io', 'quay.io']" | tee /etc/containers/registries.conf
+  # This should have been a toml_file module if I had found one
+  ini_file:
+    path: /etc/containers/registries.conf
+    section: registries.search
+    option: registries
+    value: "['docker.io', 'quay.io']"
   become: yes
-  args:
-    creates: /etc/containers
   when: ansible_facts.distribution == 'Debian'
 
+- name: Adding podman insecure registries (Debian-specific)
+  # This should have been a toml_file module if I had found one
+  ini_file:
+    path: /etc/containers/registries.conf
+    section: registries.insecure
+    option: registries
+    value: "['localhost:24817']"
+  become: yes
+  when:
+    - ansible_facts.distribution == 'Debian'
+    - pulp_webserver_disable_https | default(false)
+
 # Needed to allow non-root users to run podman containers on Debian
 - name: Enable unprivileged_userns_clone (Debian specific)
   sysctl:
     name: kernel.unprivileged_userns_clone
     value: "1"
+  become: yes
   when:
     - ansible_facts.distribution == 'Debian'
-    - ansible_facts.env.container | default("") == "podman"
+    # This variable is set in molecule/default/Dockerfile.j2
+    - ansible_facts.env.container | default("") != "docker"
```

Comment:
So we could really need a good ansible_fact that tells us whether we are in a container.

Reply:
I investigated this thoroughly around October 1st.

On Ansible 2.9:

1. `ansible_facts.env.container` only understands podman, not docker. It is either "podman", or the entire variable is undefined. (ansible_facts.env has no element container).
2. `ansible_facts.virtualization_type` understands docker, lxc, and real hypervisors.

Reply:
I thought it was tied to this line:

https://github.com/pulp/pulp_installer/blob/master/molecule/scenario_resources/Dockerfile.j2#L14

Reply:
I think we want to run it whenever we are not in a container environment.
So can we be sure, ansible sets `ansible_facts.virtualization_type` for us?
Then i would suggest:
`- ansible_facts.virtualization_type | default("") not in ["docker", "podman"]`

Reply:
And now this seems to reflect, what @fao89 realised in his ansible 2.10 pr, that `virtualization_type` does not work there.
(It is failing exactly for the combination source and 2.10)

I am out of ideas.